mapzap/models/staging/base/base_dcp__pluto_with_bbls.sql

```diff
@@ -0,0 +1,32 @@
+{% set pluto_versions_query %}
+select
+    primary_pluto_version
+from
+    {{ ref('seed_pluto_versions') }}
+where
+    has_bbl
+    and has_wkt_geometry
+order by 1
+{% endset %}
+
+{% set pluto_versions_query_result = run_query(pluto_versions_query) %}
+
+{% if execute %}
+    {# Return the first column #}
+    {% set pluto_versions = pluto_versions_query_result.columns[0].values() %}
+{% else %}
+    {% set pluto_versions = [] %}
+{% endif %}
+
+{% for pluto_table in pluto_versions %}
+
+    SELECT
+        '{{ pluto_table }}' AS pluto_version,
+        CAST(bbl AS STRING) AS bbl,
+        wkt
+    FROM {{ source('map_pluto', pluto_table) }}
+
+    {% if not loop.last %}
+        UNION ALL
```

Comment:
So this is unioning all versions of pluto? And what's the rationale for the two pluto source tables?

Reply:
this is unioning all versions that already have BBLs

the two pluto base tables are for different types of pluto versions: those that already have BBLs and those that need BBLs constructed. and then the staging table `stg_dcp__pluto_bbls` combines them for downstream use

the seed file `seed_pluto_versions.csv` (which becomes a table via `dbt seed`) stores the details for each pluto version

Reply:
Ohh so the has bbl vs needs bbl constructed is not a record-level issue but a whole table (version of pluto) level issue.

Reply:
As for unioning all versions (in the "final" staging pluto source table), how are duplicate bbls (with potentially differing geometries) dealt with? Just unioning all geometries in the aggregate step, right? Do we have any concern with using potentially outdated info? It seems like it should be fine, just sort of thinking out loud here

Reply:
ooo there's [a test for that](https://github.com/NYCPlanning/db-zap-opendata/blob/build-map-zap/mapzap/models/staging/_dcp_models.yml#L124-L127)!

duplicate BBLs in the same pluto version would be bad, so that test checks for that. but a BBL existing and/or having different geometry in a different pluto version is ok since a ZAP record only looks at a single year (aka pluto version) to get BBL geometries